### ISSUE TITLE
TestWorkerManager in `work_manager_test` is flaky.

### ISSUE THREAD

**Issue by ConnorDoyle** [intelsdi-x/snap#493]
Observed spurious assertion failures on OS X.

```go
      ...
      go manager.Work(j1)                                                                                                                                                                                                                                                             
      go manager.Work(j2)                                                                                                                                                                                                                                                             
      manager.Work(j3)                                                                                                                                                                                                                                                                
      time.Sleep(time.Millisecond * 100)                                                                                                                                                                                                                                              
                                                                                                                                                                                                                                                                                      
      worked := 0                                                                                                                                                                                                                                                                     
      for _, j := range []*mockJob{j1, j2, j3} {                                                                                                                                                                                                                                      
                                                                                                                                                                                                                                                                                      
        if j.worked == true {                                                                                                                                                                                                                                                         
          worked++                                                                                                                                                                                                                                                                    
        }                                                                                                                                                                                                                                                                             
      }                                                                                                                                                                                                                                                                               
      So(worked, ShouldEqual, 2)  // <= is sometimes 1
      ...
```

**Comment by ConnorDoyle** [comment]
Have seen this one crop up in CI a number of times lately.  Submitted #645 as a new attempt to fix this :beetle:

**Comment by ConnorDoyle** [comment]
Fixed by #645.
